src/mlpack/tests/spill_tree_test.cpp

```diff
@@ -0,0 +1,247 @@
+                node->Hyperplane().Project(node->Dataset().col(descIndex)),
+                tau);
+          }
+        }
+        if (node->Right())
+        {
+          // Let's check that points in the right child are projected to values
+          // in the range: (-tau, inf)
+          size_t numDesc = node->Right()->NumDescendants();
+          for (size_t i = 0; i < numDesc; i++)
+          {
+            size_t descIndex = node->Right()->Descendant(i);
+            BOOST_REQUIRE_GT(
+                node->Hyperplane().Project(node->Dataset().col(descIndex)),
+                -tau);
+          }
+        }
+      }
+      else
+      {
+        // We have a non-overlapping node.
+        if (node->Left())
+        {
+          // Let's check that points in the left child are considered to the
+          // left by the splitting hyperplane.
+          size_t numDesc = node->Left()->NumDescendants();
+          for (size_t i = 0; i < numDesc; i++)
+          {
+            size_t descIndex = node->Left()->Descendant(i);
+            BOOST_REQUIRE(
+                node->Hyperplane().Left(node->Dataset().col(descIndex)));
+          }
+        }
+        if (node->Right())
+        {
+          // Let's check that points in the right child are considered to the
+          // right by the splitting hyperplane.
+          size_t numDesc = node->Right()->NumDescendants();
+          for (size_t i = 0; i < numDesc; i++)
+          {
+            size_t descIndex = node->Right()->Descendant(i);
+            BOOST_REQUIRE(
+                node->Hyperplane().Right(node->Dataset().col(descIndex)));
+          }
+        }
+      }
+
+      if (node->Left())
+        nodes.push(node->Left());
+
+      if (node->Right())
+        nodes.push(node->Right());
+    }
+  }
+}
+
+/**
+ * Test to make sure that the points in the left child are considered to the
+ * left by the node's splitting hyperplane, and the same for points in the
+ * right child.
+ */
+BOOST_AUTO_TEST_CASE(SpillTreeHyperplaneTest)
+{
+  typedef SPTree<EuclideanDistance, EmptyStatistic, arma::mat> SpillType1;
+  typedef NonOrtSPTree<EuclideanDistance, EmptyStatistic, arma::mat> SpillType2;
+  typedef MeanSPTree<EuclideanDistance, EmptyStatistic, arma::mat> SpillType3;
+  typedef NonOrtMeanSPTree<EuclideanDistance, EmptyStatistic, arma::mat>
+      SpillType4;
+
+  SpillTreeHyperplaneTestAux<SpillType1>();
+  SpillTreeHyperplaneTestAux<SpillType2>();
+  SpillTreeHyperplaneTestAux<SpillType3>();
+  SpillTreeHyperplaneTestAux<SpillType4>();
+}
+
+/**
+ * Simple test for the move constructor.
+ */
+BOOST_AUTO_TEST_CASE(SpillTreeMoveDatasetTest)
+{
+  arma::mat dataset = arma::randu<arma::mat>(3, 1000);
+  typedef SPTree<EuclideanDistance, EmptyStatistic, arma::mat> TreeType;
+
+  TreeType tree(std::move(dataset));
+
+  BOOST_REQUIRE_EQUAL(dataset.n_elem, 0);
+  BOOST_REQUIRE_EQUAL(tree.Dataset().n_rows, 3);
+  BOOST_REQUIRE_EQUAL(tree.Dataset().n_cols, 1000);
+}
+
```

Comment:
Ok. Done in: https://github.com/mlpack/mlpack/pull/747/commits/f1253d37af50ab7df14a2c841c0a7345cec140e1